homeassistant/components/govee_light_local/coordinator.py

```diff
@@ -32,32 +34,45 @@ def __init__(self, hass: HomeAssistant) -> None:
             update_interval=SCAN_INTERVAL,
         )
 
-        self._controller = GoveeController(
-            loop=hass.loop,
-            logger=_LOGGER,
-            broadcast_address=CONF_MULTICAST_ADDRESS_DEFAULT,
-            broadcast_port=CONF_TARGET_PORT_DEFAULT,
-            listening_port=CONF_LISTENING_PORT_DEFAULT,
-            discovery_enabled=True,
-            discovery_interval=CONF_DISCOVERY_INTERVAL_DEFAULT,
-            discovered_callback=None,
-            update_enabled=False,
-        )
+        self._controllers = []
+        for source_ip in source_ips:
+            self._controllers.append(
+                GoveeController(
+                    loop=hass.loop,
+                    logger=_LOGGER,
+                    listening_address=str(source_ip),
+                    broadcast_address=CONF_MULTICAST_ADDRESS_DEFAULT,
+                    broadcast_port=CONF_TARGET_PORT_DEFAULT,
+                    listening_port=CONF_LISTENING_PORT_DEFAULT,
+                    discovery_enabled=True,
+                    discovery_interval=1,
+                    update_enabled=False,
+                )
+            )
 
     async def start(self) -> None:
         """Start the Govee coordinator."""
-        await self._controller.start()
-        self._controller.send_update_message()
+
+        for controller in self._controllers:
+            await controller.start()
+            controller.send_update_message()
 
     async def set_discovery_callback(
         self, callback: Callable[[GoveeDevice, bool], bool]
     ) -> None:
         """Set discovery callback for automatic Govee light discovery."""
-        self._controller.set_device_discovered_callback(callback)
 
-    def cleanup(self) -> asyncio.Event:
+        for controller in self._controllers:
+            controller.set_device_discovered_callback(callback)
+
+    def cleanup(self) -> list[asyncio.Event]:
         """Stop and cleanup the cooridinator."""
-        return self._controller.cleanup()
+
+        results: list[asyncio.Event] = []
```

Comment:
This seems like a good candidate for list comprehension

Reply:
done.